experimental/AlgebraicShifting/src/UniformHypergraph.jl

```diff
@@ -0,0 +1,87 @@
+# this might not be necessary if we can use polymake hasse diagram functionality?
+function complex_faces(K :: SimplicialComplex, d :: Int) :: Vector{Vector{Int}}
+  return sort(union(subsets.(sort.(collect.(facets(K))), d+1)...))
+  # return union(subsets.(facets(K), d+1)...)
+end
+
+struct UniformHypergraph
+  n_vertices::Int
+  k::Int
+  faces::Vector{Vector{Int}}
+end
+
+function uniform_hypergraph(faces::Vector{Vector{Int}}, n::Int, k::Int)
```

Comment:
This basically does some sanity checking and normalization. Do things like this belong in an internal constructor? Don't know OSCAR's conventions here.

Reply:
Invariants that are supposed to be kept by all instances of an object generally belong into an (inner) constructor, yes.